packages/realm/src/Object.ts

```diff
@@ -120,13 +122,52 @@ class RealmObject<T = DefaultObject> {
   entries(): [string, unknown][] {
     throw new Error("Not yet implemented");
   }
-  toJSON(): unknown {
-    // return { ...this };
+
+  toJSON(_?: string, cache: JSONCacheMap = new Map()): DefaultObject {
+    // Construct a reference-id of table-name & primaryKey if it exists, or fall back to objectId.
+
+    // Check if current objectId has already processed, to keep object references the same.
+    const existing = this._getOwnCache(cache);
+    if (existing) {
+      return existing;
+    }
+
+    // Create new result, and store in cache.
+    const result: DefaultObject = {};
+    this._setOwnCache(cache, result);
+
+    // Move all enumerable keys to result, triggering any specific toJSON implementation in the process.
+    // TODO: Consider Object.keys(this)
```

Comment:
is there any value to `.key(..).concat(...prototype)` now? again I skipped as when I started only looping was possible so I am unsure whether this is not equivalent with the concat.